lib/mllp/receiver.ex

```diff
@@ -62,10 +84,74 @@ defmodule MLLP.Receiver do
   @spec stop(any) :: :ok | {:error, :not_found}
   def stop(port) do
     receiver_id = get_receiver_id_by_port(port)
-
     :ok = :ranch.stop_listener(receiver_id)
   end
 
+  @doc """
+  A function which can be used to embed an MLLP.Receiver under Elixir v1.5+ supervisors.
+  Unlike `start/1`, `start/2`, or `start/3` this function takes two additional options : `ref` and `transport_opts`.
+  Note that if a `ref` option is not supplied a reference will be created for you using `make_ref/0`.
+      children = [{MLLP.Receiver, [
+          ref: MyRef,
+          port: 4090,
+          dispatcher: MyDispatcher,
+          packet_framer: MyPacketFramer,
+          transport_opts: %{num_acceptors: 25, max_connections: 20_000}
+        ]}
+      ]
+      Supervisor.init(children, strategy: :one_for_one)
+  See [Options](#module-options) for details on required and optiomal parameters.
+  ## Examples
+      iex(1)> opts = [ref: MyRef, port: 4090, dispatcher: MyDispatcher, packet_framer: MyPacketFramer]
+      [
+        ref: MyRef,
+        port: 4090,
+        dispatcher: MyDispatcher,
+        packet_framer: MyPacketFramer
+      ]
+      iex(2)> MLLP.Receiver.child_spec(opts)
+      %{
+        id: {:ranch_listener_sup, MyRef},
+        modules: [:ranch_listener_sup],
+        restart: :permanent,
+        shutdown: :infinity,
+        start: {:ranch_listener_sup, :start_link,
+         [
+          MyRef,
+          :ranch_tcp,
+          %{socket_opts: [port: 4090]},
+          MLLP.Receiver,
+          [packet_framer_module: MyPacketFramer, dispatcher_module: MyDispatcher]
+        ]},
+        type: :supervisor
+      }
+  """
+  @spec child_spec(options()) :: Superviosr.child_spec()
+  def child_spec(opts) do
+    args = to_args(opts)
+
+    res =
+      :ranch.child_spec(
+        args.receiver_id,
+        args.transport_mod,
+        args.transport_opts,
+        args.proto_mod,
+        args.proto_opts
+      )
+
+    case res do
+      {id, start, restart, shutdown, type, modules} ->
```

Comment:
Do we really need this case, we can use pattern match on line 133

Reply:
Do you mean ... 

`case :ranch.child_spec(...) do  ...` ? 

If so, if I can't fit something into a case expression cleanly, I opt to assign to a variable. I try not to look at variables as a  cost but as a chance to provide the reader more context. In that sense, `res` is a bad name here. wdyt?

Reply:
Oh, I think you mean  why not just do `{id, start, restart, shutdown, type, modules} = :ranch.child_spec(...)` 

It's because it line wrapped, but I have no  strong feelings and will adjust.

Reply:
As there is no other case condition, I was thinking like this

```
{id, start, restart, shutdown, type, modules} = :ranch.child_spec(
        args.receiver_id,
        args.transport_mod,
        args.transport_opts,
        args.proto_mod,
        args.proto_opts
      )
```

Reply:
Done.